### PR TITLE
[Model Monitoring] Add endpoint registration to v2 serving + config changes

### PR DESCRIPTION
**Pull request by Michaelliv**
* Updated model monitoring config to allow a similar functionality to the feature store config, this PR introduces a single template for data store paths instead of individual paths for kv, tsdb and parquets. 
* Implemented endpoint "registration" in v2_model_server.

### CODE REVIEW

**Review by Hedingber — APPROVED** [review]

File: mlrun/api/crud/model_endpoints.py [hunk 1]
```diff
@@ -456,6 +481,12 @@ def build_kv_cursor_filter_expression(
     return " AND ".join(filter_expression)
 
 
+def parse_store_prefix(store_prefix: str):
+    scheme, path = store_prefix.split("///", 1)
```
Comment (yaronha) [inline mlrun/api/crud/model_endpoints.py:485]: should use the same func as we use in feature store (w url parse and extract endpoint as well)

File: mlrun/serving/server.py [hunk 2]
```diff
@@ -39,22 +39,30 @@ class _StreamContext:
...
+
+        if ((enabled and stream_uri) or log_stream) and function_uri:
             self.enabled = True
-            log_stream = log_stream or stream_url
+
             project, _, _, _ = parse_versioned_object_uri(
                 function_uri, config.default_project
             )
+
+            if stream_uri:
```
Comment (yaronha) [inline mlrun/serving/server.py:56]: isnt it always True?

File: mlrun/serving/v2_serving.py [hunk 3]
```diff
@@ -103,6 +110,11 @@ def post_init(self, mode="sync"):
             else:
                 self._load_and_update_state()
 
+        if self._model_logger is not None and not isinstance(
+            self._model_logger.output_stream, _DummyStream
```
Comment (yaronha) [inline mlrun/serving/v2_serving.py:114]: better move the condition to the `init_endpoint_record` func

File: mlrun/serving/v2_serving.py [hunk 4]
```diff
@@ -299,6 +312,35 @@ def __init__(self, model, context, output_stream=None):
         self._batch_iter = 0
         self._batch = []
 
+    def init_endpoint_record(self):
+        project, uri, tag, hash_key = parse_versioned_object_uri(self.function_uri)
```
Comment (yaronha) [inline mlrun/serving/v2_serving.py:316]: what about exception handling, e.g. v3io doesnt exist (in open source) or IO error..

**Pull request merged**